compiler/src/iree/compiler/Codegen/Common/GPU/GPUHeuristics.cpp

```diff
@@ -132,122 +131,261 @@ FailureOr<GPUMMASchedule> fitScheduleInSharedMemory(
     // If no dimension can be shrunk, give up.
     return failure();
   }
+
+  LLVM_DEBUG({
+    llvm::dbgs() << "Chosen schedule is valid:\n";
+    llvm::dbgs() << "mSize: " << schedule.mSize << "\n";
```

Comment:
At this point it may make sense to add a `custom operator<<` for the schedule struct

Reply:
done